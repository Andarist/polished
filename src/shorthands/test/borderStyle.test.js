// @flow
import borderStyle from '../borderStyle'

describe('borderStyle', () => {
  beforeAll(() => {
    global.console = {
      error: jest.fn(),
      warn: jest.fn(),
      log: global.console.log,
    }
  })

  afterEach(() => {
    // eslint-disable-next-line no-console
    console.error.mockClear()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })

  it('properly applies a value when passed only one', () => {
    expect(borderStyle('solid')).toMatchSnapshot()
  })
  it('properly applies values when passed two', () => {
    expect(borderStyle('solid', 'dashed')).toMatchSnapshot()
  })
  it('properly applies values when passed three', () => {
    expect(borderStyle('solid', 'dashed', 'dotted')).toMatchSnapshot()
  })
  it('properly applies values when passed four', () => {
    expect(borderStyle('solid', 'dashed', 'dotted', 'double')).toMatchSnapshot()
  })
  it('throws an error for each invalid value', () => {
    borderStyle('top', 'right', 'bottom', 'left')
    // eslint-disable-next-line no-console
    expect(console.error.mock.calls).toMatchSnapshot()
  })
  it('throws a warning when passed too many parameters', () => {
    borderStyle('solid', 'dashed', 'dotted', 'double', 'none')
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
  })
})
