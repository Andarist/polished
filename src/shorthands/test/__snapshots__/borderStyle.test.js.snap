exports[`borderStyle properly applies a value when passed only one 1`] = `
Object {
  "borderBottomStyle": "solid",
  "borderLeftStyle": "solid",
  "borderRightStyle": "solid",
  "borderTopStyle": "solid",
}
`;

exports[`borderStyle properly applies values when passed four 1`] = `
Object {
  "borderBottomStyle": "dotted",
  "borderLeftStyle": "double",
  "borderRightStyle": "dashed",
  "borderTopStyle": "solid",
}
`;

exports[`borderStyle properly applies values when passed three 1`] = `
Object {
  "borderBottomStyle": "dotted",
  "borderLeftStyle": "dashed",
  "borderRightStyle": "dashed",
  "borderTopStyle": "solid",
}
`;

exports[`borderStyle properly applies values when passed two 1`] = `
Object {
  "borderBottomStyle": "solid",
  "borderLeftStyle": "dashed",
  "borderRightStyle": "dashed",
  "borderTopStyle": "solid",
}
`;

exports[`borderStyle throws a warning when passed too many parameters 1`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- shorthands/borderStyle.js --%c

borderStyle %cexpects a maximum of %c4 parameters%c. However, you passed %c5 parameters%c. This additional parameter was ignored.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#borderStyle%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: green",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: goldenrod",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;

exports[`borderStyle throws an error for each invalid value 1`] = `
Array [
  Array [
    "%c ✨ ERROR ✨ ---- shorthands/borderStyle.js --%c

borderStyle %cexpects a %c1st parameter%c(%cfirstStyle%c: %ccssBorderStyle%c). However, you passed (%c\'top\'%c:%cstring%c) instead.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#borderStyle%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
  Array [
    "%c ✨ ERROR ✨ ---- shorthands/borderStyle.js --%c

borderStyle %cexpects a %c2nd parameter%c(%csecondStyle%c: %ccssBorderStyle%c). However, you passed (%c\'right\'%c:%cstring%c) instead.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#borderStyle%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
  Array [
    "%c ✨ ERROR ✨ ---- shorthands/borderStyle.js --%c

borderStyle %cexpects a %c3rd parameter%c(%cthirdStyle%c: %ccssBorderStyle%c). However, you passed (%c\'bottom\'%c:%cstring%c) instead.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#borderStyle%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
  Array [
    "%c ✨ ERROR ✨ ---- shorthands/borderStyle.js --%c

borderStyle %cexpects a %c4th parameter%c(%cfourthStyle%c: %ccssBorderStyle%c). However, you passed (%c\'left\'%c:%cstring%c) instead.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#borderStyle%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: green; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: red; font-size: 12px; font-style: italic",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;
